Add server log page

lcsfelix opened the pull request.

RafaelTaranto committed:
Lamassu admin server initial commit

RafaelTaranto committed:
Lamassu admin server initial commit

lcsfelix committed:
Add server log page

lcsfelix committed:
Add server log page

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.js:
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { useSelect } from 'downshift'
+import _ from 'lodash/fp'
```
You can change this to only import what you need, check out the other places where I'm using lodash. It's better for the final bundle size.

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.js:
```diff
@@ -0,0 +1,60 @@
+
+  return (
+    <div className={classnames(selectClassNames)}>
+      <label {...getLabelProps()}>Level</label>
```
Are we using the label?

lcsfelix replied:
We are. I just realized it should probably be passed into the component, though, instead of a fixed value.

![image](https://user-images.githubusercontent.com/5789328/67568765-fef71600-f724-11e9-9ebc-d94fdbd9e22f.png)

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.js:
```diff
@@ -0,0 +1,60 @@
+      <label {...getLabelProps()}>Level</label>
+      <button
+        {...getToggleButtonProps()}
+        style={isOpen ? { borderRadius: '16px 16px 0 0' } : {}}
```
This should be in the styles and applied conditionally using classnames.

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.js:
```diff
@@ -0,0 +1,60 @@
+        {...getToggleButtonProps()}
+        style={isOpen ? { borderRadius: '16px 16px 0 0' } : {}}
+      >
+        {_.capitalize(selectedItem)} <Arrowdown />
```
Why is captalize being used here?

RafaelTaranto replied:
Oh, ok. You're getting the log levels from the response. Lodash has another function that might be better for this. https://lodash.com/docs/4.17.15#startCase

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.js:
```diff
@@ -0,0 +1,60 @@
+              key={`${item}${index}`}
+              {...getItemProps({ item, index })}
+            >
+              {_.capitalize(item)}
```
Why captalize?

RafaelTaranto replied:
Same as above

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.styles.js:
```diff
@@ -0,0 +1,72 @@
+      width: `${WIDTH}px`,
+      padding: '6px 0 6px 12px',
+      borderRadius: '20px',
+      fontFamily: 'MuseoSans',
```
We have well defined font styles, it's better if you extend from a predefined one. Take a look at other styles files using extends.

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.styles.js:
```diff
@@ -0,0 +1,72 @@
+
+export default {
+  select: {
+    width: `${WIDTH}px`,
```
Whenever the value of something is `x` pixels you can do
```
{
  width: 200
}
```

You should replace this across this file wherever you can.

RafaelTaranto commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react'
+import _ from 'lodash/fp'
```
Only import what you need from lodash

lcsfelix committed:
fix: select component

lcsfelix committed:
fix: select component

RafaelTaranto commented on new-lamassu-admin/src/components/inputs/Select.styles.js:
```diff
@@ -0,0 +1,77 @@
+      borderRadius: 20,
+      fontFamily: fontSecondary,
+      fontSize: 18,
+      fontWeight: 500,
```
Still not what I meant, doesn't this equal some set typography in typography.js? It will probably be some set font style already. like tl2, tl1, or info1... Search for `extend:` on the codebase to see the use

lcsfelix committed:
feat: add server version

lcsfelix committed:
feat: add server version

lcsfelix committed:
fix: use defined font styles

lcsfelix committed:
fix: use defined font styles

lcsfelix committed:
fix: style select label

lcsfelix committed:
fix: style select label

lcsfelix committed:
feat: add uptime dashboards

note: requires supervisord running

lcsfelix committed:
feat: add uptime dashboards

note: requires supervisord running

lcsfelix committed:
feat: style download and share buttons

lcsfelix committed:
feat: style download and share buttons

joshmh commented on lib/new-admin/admin-server.js:
```diff
@@ -56,6 +57,42 @@ app.post('/api/support_logs', (req, res, next) => {
+        }
+      })
+    )
+  })
```
This kind of detailed stuff, including bringing in dependencies like ``xmlrpc`` should be done in a separate supervisor-specific module.

lcsfelix replied:
Is https://github.com/RafaelTaranto/lamassu-server/pull/1/commits/85a3a05275a160e36582a90a7349f16810be6983 what you mean?

joshmh replied:
Yep.

joshmh commented on lib/new-admin/admin-server.js:
```diff
@@ -56,6 +57,42 @@ app.post('/api/support_logs', (req, res, next) => {
+      return 'FATAL'
+    } else {
+      return state
+    }
```
This would be cleaner:

```
    if (state === 'EXITED' || state === 'STOPPING') return 'STOPPED'
    if (state === 'STARTING') return 'RUNNING'
    if (state === 'BACKOFF') return 'FATAL'
    return state
```

Or, a switch statement would work well here.

joshmh commented on new-lamassu-admin/src/components/LogPageHeader.js:
```diff
@@ -0,0 +1,104 @@
+      display: 'none'
+    }
+  }
+}
```
Could this be defined in a more general module for the whole project?

joshmh commented on new-lamassu-admin/src/components/Uptime.js:
```diff
@@ -0,0 +1,69 @@
+import typographyStyles from './typography/styles'
+const { label } = typographyStyles
+
+// import styles from './Uptime.styles'
```
Why is this commented out? If you don't need it you should delete it.

joshmh commented on new-lamassu-admin/src/components/Uptime.js:
```diff
@@ -0,0 +1,69 @@
+      return `${floor(time / 60 / 60 / 24, 0)}d`
+    }
+  }
+
```
Same as before, no need for ``else``s here.

joshmh commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,181 @@
+
+const { regularLabel } = typographyStyles
+
+// import styles from './Logs.module.scss'
```
Remove comment.

joshmh commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,181 @@
+  const classes = useStyles()
+
+  useAxios({
+    url: 'http://localhost:8070/api/version',
```
Port and hostname shouldn't be hardcoded.

joshmh commented on new-lamassu-admin/src/routing/routes.js:
```diff
@@ -17,7 +18,8 @@ const tree = [
     children: [
       { key: 'logs', label: 'Logs', route: '/maintenance/logs' },
-      { key: 'fuding', label: 'Funding', route: '/maintenance/funding' }
+      { key: 'fuding', label: 'Funding', route: '/maintenance/funding' },
```
Typo: should be "funding".

lcsfelix committed:
fix: replace if-else with switch

lcsfelix committed:
fix: replace if-else with switch

lcsfelix committed:
fix: refactor SimpleButton CSS

lcsfelix committed:
fix: refactor SimpleButton CSS

lcsfelix committed:
fix: remove unused commented imports

lcsfelix committed:
fix: remove unused commented imports

lcsfelix committed:
fix: move supervisor logic to own module

lcsfelix committed:
fix: move supervisor logic to own module

RafaelTaranto commented on lib/new-admin/supervisor.js:
```diff
@@ -0,0 +1,35 @@
+  const convertStates = (state) => {
+    switch (state) {
+      case 'EXITED':
+        return 'STOPPED'
```
You can remove this line

RafaelTaranto replied:
I think it's better if you also make all possible states defined here.

RafaelTaranto commented on lib/new-admin/supervisor.js:
```diff
@@ -0,0 +1,35 @@
+    port: '9001',
+    path: '/RPC2'
+  }).methodCall('supervisor.getAllProcessInfo', [], function (error, value) {
+    if (error) console.log(error) // TODO: properly log error
```
This should use logger

RafaelTaranto commented on lib/new-admin/admin-server.js:
```diff
@@ -56,6 +57,15 @@ app.post('/api/support_logs', (req, res, next) => {
+  res.send(require('../../package.json').version)
+})
+
+// TODO: get URL from config file somewhere
```
I don't get this TODO.

lcsfelix replied:
I don't either. It's been removed now

RafaelTaranto commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,180 @@
+      }
+      setSaveMessage('✓ Saved latest snapshot')
+    }
+  })
```
This looks like the one from machine, we don't have device_id for this one, and the support logs db is different for lamassu-server logs

RafaelTaranto commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,180 @@
+    customHandler: () => {
+      setSaveMessage('')
+    }
+  })
```
Looks like the same from lamassu-machine with deviceId.

lcsfelix replied:
It is, this is pre-implementation of the logger. I kept all the machine related stuff to help the design process.

RafaelTaranto commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,180 @@
+        setSelected(get('data.machines[0]')(res))
+      }
+    }
+  })
```
We don't need machines on this page

RafaelTaranto commented on new-lamassu-admin/src/pages/ServerLogs.js:
```diff
@@ -0,0 +1,180 @@
+}
+
+const Logs = () => {
+  const [machines, setMachines] = useState(null)
```
This is not being used

RafaelTaranto commented on package.json:
```diff
@@ -7,6 +7,7 @@
   "author": "Lamassu (https://lamassu.is)",
   "dependencies": {
     "@fczbkk/uuid4": "^3.0.0",
+    "@pauleliet/winston-pg-native": "^4.4.0",
```
You can remove this

lcsfelix committed:
feat: log server messages to db

lcsfelix committed:
fix: change server log table width

lcsfelix committed:
fix: promisify supervisor module

lcsfelix committed:
fix: define all supervisord states

lcsfelix committed:
fix: increase select component z-index

lcsfelix committed:
fix: remove winston-pg-native from package.json

lcsfelix committed:
Merge branch 'new-lamassu-admin' of github.com:RafaelTaranto/lamassu-server into new-server-logs

lcsfelix committed:
Merge branch 'new-lamassu-admin' of github.com:RafaelTaranto/lamassu-server into new-server-logs

lcsfelix committed:
fix: remove LogPageHeader component

Replaced with a JSS stylesheet, as functionality of buttons differs
between machine and server pages.

lcsfelix committed:
fix: add winston-transport to package

lcsfelix committed:
fix: package-lock conflicts

joshmh commented on lib/new-admin/supervisor.js:
```diff
@@ -1,35 +1,57 @@
+        return 'FATAL'
       default:
-        return state
+        return 'FATAL'
```
Should also log that we've seen an unsupported state, in case of ``default``.

lcsfelix replied:
Resolved in https://github.com/RafaelTaranto/lamassu-server/pull/1/commits/ef399b0987842b56ceaf31347a40b4ecfa356e69.

lcsfelix committed:
fix: log unexpected supervisord state
